google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurator.java

```diff
@@ -52,20 +56,45 @@ public AppEngineDeploymentConfigurator(Project project) {
   @NotNull
   @Override
   public List<DeploymentSource> getAvailableDeploymentSources() {
-    List<DeploymentSource> deploymentSources = new ArrayList<DeploymentSource>();
+    List<DeploymentSource> deploymentSources = new ArrayList<>();
+
+    for (Module module : ModuleManager.getInstance(project).getModules()) {
+      if (isJarOrWarMavenBuild(module)) {
+        deploymentSources.add(
+            new MavenBuildDeploymentSource(
+                ModulePointerManager.getInstance(project).create(module), project));
+      }
+    }
 
     ModulePointer modulePointer =
         ModulePointerManager.getInstance(project).create("userSpecifiedSource");
     deploymentSources.add(new UserSpecifiedPathDeploymentSource(modulePointer));
 
     deploymentSources.addAll(JavaDeploymentSourceUtil
-        .getInstance().createArtifactDeploymentSources(project, getJarsAndWars()));
+        .getInstance().createArtifactDeploymentSources(project, getJarAndWarArtifacts()));
 
     return deploymentSources;
   }
 
-  private List<Artifact> getJarsAndWars() {
-    List<Artifact> jarsAndWars = new ArrayList<Artifact>();
+  private boolean isJarOrWarMavenBuild(Module module) {
+    boolean isMavenProject = MavenProjectsManager.getInstance(project).isMavenizedModule(module)
```

Comment:
it looks like we call MoavenPRojectsManager.getInstance(project) a lot in this method, what do you think about extracting a variable?